latex/Inhalt/grundlagen.tex

```diff
@@ -0,0 +1,193 @@
+
+\begin{minted}{rust}
+    match x {
+        1 => /* x == 1... */,
+        2..=5 => /* x in [2,3,4,5] */,
+        ...
+        _ => // default case
+    }
+\end{minted}
+
+Rust garantiert zu Compilezeit, dass die match-Arme alle möglichen Fälle abdecken (nur relevant wenn kein default-Case).
+
+\subsection{Result und Option}
+
+Rust hat kein \code{null}, stattdessen existiert der \rust{Option<T>} Typ. Option ist wie folgt definiert:
+
+\begin{minted}{rust}
+    enum Option<T> {
+        Some(T),
+        None
+    }
+\end{minted}
+
+Ein Element vom Typ \rust{Option<i32>} enthält also entweder nichts (None), oder oder eine 32-Bit Integer (Some(i32)). Um an den enthaltenden Wert zu kommen, muss eine Fallunterscheidung durchgeführt werden, bspw. durch ein \rust{match}-Statement. Dies garantiert, dass keine ungewollten null-References möglich sind (wie \zB ein Methodenaufruf auf \code{null}).
+
+\rust{Option<T>} wird verwendet, wenn es akzeptabel ist, dass kein Wert vorhanden ist. Andernfalls sollte \rust{Result<T, E>} verwendet werden.
+
+\begin{minted}{rust}
+    enum Result<T, E> {
+        Ok(T),
+        Err(E)
+    }
+\end{minted}
+
+Ein Result enthält entweder einen Wert des entsprechenden Types, oder einen Error mit einer Error-Message des entsprechenden Typs (oft ein String).
+
+
+\section{WebAssembly}
+
+"WebAssembly is a safe, portable, low-level code format designed for efficient execution and compact representation"\cite{WebAssemblyCoreSpecification}. Im Endeffekt handelt es sich bei WebAssembly also um eine low-level Bytecode-Sprache, die von Browsern ausgeführt werden kann. Diese Sprache soll ähnlich performant sein, wie die Ausführung naives Maschinen-Codes. Das Paper, in dem WebAssembly eingeführt wird, berichtet eine 10\% Performance-Diskrepanz zwischen WASM und naivem Assembly\cite{10.1145/3062341.3062363}. Durch Kompilation nach WebAssembly ist es möglich Programme auf Seite des Clienten laufen zu lassen, die sonst vom Server ausgeführt werden müssten.
+Für viele Programmiersprachen gibt es entsprechende Compiler, die es ermöglichen nach WebAssembly zu übersetzen (bspw. C/C++, Rust, u.s.w).
+
+In der entsprechenden Sprache muss explizit die Schnittstelle zu JavaScript deklariert werden, um festzulegen welche Funktionalitäten dem Frontend zur Verfügung stehen.
+
+\section{Intel 8080}
+
+Das Folgende ist eine allgemeine Übersicht über die Architektur und Funktionsweise einer Intel 8080 CPU. Die Informationen dem offiziellen Datenblatt\cite{datasheet} und dem offizielen Programmierhandbuch\cite{progManual} gewonnen.
+Einige Aspekte --- \zB die Taktung des Intel 8080 --- welche nicht emuliert werden müssen, werden bewusst nicht erwähnt. Die Übersicht soll ausschließlich Details erläutern, die relevant zum Verständnis unseres Emulators sind.
+
+\subsubsection{RAM und Stack}
+
+Im RAM (Arbeitsspeicher) liegt sowohl der Programmcode, als auch der Stack.
```

Comment:
Again, abbrevations need to be written out before using them if I am not mistaken -> Which we do in the following lines , without using a package for that :(